code/drasil-example/template/lib/Drasil/Template/Body.hs

```diff
@@ -141,3 +181,191 @@ progName = commonIdeaWithDict "progName" (pn "ProgName") "ProgName" []
+equilibriumQ = mkQuant' -- A somewhat odd variant of `mkQuant` that re-orders argument and adds staging to the `Symbol`
+  "inEquilibrium"
+  (nounPhrase "in equilibrium" "all equilibria") -- duplicates the IdeaDict
+  Nothing -- UnitDefn
+  Boolean
+  (autoStage $ label "inEquilibrium")
+  Nothing -- "abbreviation String"
+
+price :: IdeaDict -- I guess making this is completely unnecessary because the QuantityDict will make its own.
+price = nc "price" $ cn "price"
+
+priceQ :: QuantityDict
+priceQ = mkQuant'
+  "priceQuant" -- duplicates the IdeaDict
+  (cn "price")
+  Nothing -- UnitDefn // FIXME: Uh oh! There's seemingly no way we can define a new unit: $CAD.s
+  Real -- Not quite a "Real," more of a "dollar amount with two decimal places, or a tuple containing two integers, with the second capped at 100"
+  (autoStage cP)
+  Nothing -- Abbreviation
+
+supply :: IdeaDict
+supply = nc "supply" $ cnIES "supply"
+
+supplyDemandFuncSpace :: Space
+supplyDemandFuncSpace = mkFunction [Real] Integer -- Real should be Dollar, output should be strictly non-negative integers (Z^+)
+
+supplyQ :: QuantityDict
+supplyQ = mkQuant'
+  "supplyQuant"
+  (cnIES "supply")
+  Nothing -- UnitDefn -- Should units be a part of expressions? Supply is a function, so it doesn't have a unit. However, the output of the function should have a unit. The input of the function should have units as well.
+  supplyDemandFuncSpace
+  (autoStage cS)
+  Nothing -- Abbreviation
+
+demand :: IdeaDict
+demand = nc "demand" $ cnIES "demand"
+
+demandQ :: QuantityDict
+demandQ = mkQuant'
+  "demandQuant"
+  (cnIES "demand")
+  Nothing -- UnitDefn
+  supplyDemandFuncSpace
+  (autoStage cD)
+  Nothing -- Abbreviation
+
+equilibriumQD :: QDefinition ModelExpr -- equilibrium = S(P) == D(P)
+equilibriumQD = mkQuantDef equilibriumQ $ apply1 supplyQ priceQ $= apply1 demandQ priceQ
+
+equilibriumTM :: TheoryModel
+equilibriumTM = tmNoRefs
+  (equationalModel' equilibriumQD)
+  [qw demandQ, qw supplyQ, qw priceQ]
+  ([] :: [ConceptChunk]) -- This looks like just a wart? IIRC, no tmNoRefs/tm constructor uses this parameter
+  [equilibriumQD] -- defined quantities
+  [] -- defined invariants
+  [] -- defined functions
+  "equilibriumTM"
+  [S "Equilibrium occurs when the supply and demand curves intersect at current price."]
+
+-- Note: In the above, the label for "equilibrium" (the QuantityDict) is only
+-- assigned in the TheoryModel!
+
+applePriceQ :: QuantityDict
+applePriceQ = mkQuant'
+  "applePriceQ"
+  (cn "apple price")
+  Nothing -- UnitDefn
+  Real -- Space
+  (autoStage $ sub lP cA)
+  Nothing -- Abbreviation
+
+linearSupplyQ :: QuantityDict
+linearSupplyQ = mkQuant'
+  "linearSupplyQuant"
+  (pn "price-dependant apple supply")
+  Nothing -- UnitDefn
+  (mkFunction [Real] Integer)
+  (autoStage $ sub cS lL)
+  Nothing -- Abbreviation
+
+linearDemandQ :: QuantityDict
+linearDemandQ = mkQuant'
+  "linearDemandQuant"
+  (pn "price-dependant apple demand")
```

Comment:
here all the `pn`s below: I think you're abusing `pn` here, a different constructor should be used.

Also, these 'noun phrases' should be built compositionally! The word 'apple' should never be repeated or even mentioned in all of thse (in the source). Similarly, `demand`, `marginal`, `supply` and `price` should all be abstracted.